mvlearn/semi_supervised/ctregression.py

```diff
@@ -0,0 +1,550 @@
+    >>> ctr = ctr.fit([X1, X2], y_train)
+    >>> pred = ctr.predict([X1, X2])
+    >>> print("True value\n{}".format(y))
+    True value
+    [10, 11, 12, 13, 14, 15, 16]
+    >>> print("Predicted value\n{}".format(pred))
+    Predicted value
+    [10.75 11.25 11.25 13.25 13.25 14.75 15.25]
+
+    Notes
+    -----
+    Multi-view co-training is most helpful for tasks in semi-supervised
+    learning where each view offers unique information not seen in the
+    other. As is shown in the example notebooks for using this algorithm,
+    multi-view co-training can provide good regression results even
+    when number of unlabeled samples far exceeds the number of labeled
+    samples. This regressor uses 2 sklearn regressors which work individually
+    on each view but which share information and thus result in improved
+    performance over looking at the views completely separately.
+    The regressor needs to be KNeighborsRegressor,
+    as described in the [#1CTR]_.
+
+    Algorithm
+    ---------
+    Given:
+
+        * a set *L1*, *L2* having labeled training
+        samples of each view respectively
+
+        * a set *U* of unlabeled samples
+
+        Create a pool *U'* of examples by choosing examples at random
+        from *U*
+
+        * Use *L1* to train a regressor *h1* (``estimator1``) that considers
+          only the view1 portion of the data (i.e. Xs[0])
+        * Use *L2* to train a regressor *h2* (``estimator2``) that considers
+          only the view2 portion of the data (i.e. Xs[1])
+
+        Loop for *T* iterations
+            * for each view *j*
+                * for each *u* in *U'*
+                    * Calculate the neighbors of *u*
+                    * Predict the value of *u* using *hj* estimator
+                    * create a new estimator *hj'* with same parameters
+                    as that of *hj* and train it on the data (*Lj* union *u*)
+                    * predict the value of neighbors from estimator *hj*
+                    and calculate the mean squared error with respect to
+                    original values
+                    * predict the value of neighbors from the new
+                    estimator *hj'* and calculate the mean squared error
+                    with respect to original values
+                    * calculate the difference between the two errors
+                    * store the error in a list named *deltaj*
+            * select the index with maximum positive value from both
+            the *delta1* and *delta2*
+            * let the indexes selected be *index1* and *index2*
+            * Add the *index1* to *L2*
+            * Add the *index2* to *L1*
+            * Remove the  selected index from *U'* and replenish
+            it by taking unlabeled index from *U*
+            * Use *L1* to train the regressor *h1*
+            * Use *L2* to train the regressor *h2*
+
+    References
+    ---------
```

Comment:
Sorry I should have noted this too. You need 1 more dash here (and also under Attributes) so they're the same length as the word